File: c#_labs/Lab3/GameStarter.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Lab3
{
    public class GameStarter
    {
        // Private variables
        private State state;
        private String input;

        // Constructor
        public GameStarter(String input)
        {
            this.input = input;
        }

        // Start the game
        // We first validate the data and then choose the appropriate state
        public void StartGame()
        {
            // Validate input etc
            ValidateInput validateInput = new ValidateInput();
            bool inputIsValid = false;
            try
            {
                inputIsValid = validateInput.InputIsValid(input);
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
            if (inputIsValid)
                this.state = new GameLoopState();
            else
                this.state = new InvalidGameState();

            HandleGameState();
        }

        // Start handling the game state
        private void HandleGameState()
        {
            state.HandleGameState(input, this);
        }

        // Change the game state
        public void ChangeState(String input, State state)
        {
            this.input = input;
            this.state = state;
            HandleGameState();
        }

        // Fetch the current state
        public State GetState()
        {
            return state;
        }
    }
}

File: c#_labs/Lab3_c++_state_backup/GameStarter.cpp
#include <iostream>
#include "GameStarter.h"
#include <string>
#include "Game.h"
#include "ConcreteGameStates.h"

using namespace std;

//class GameState;
Game game;

GameStarter::GameStarter(char* newInput)
{
	input = newInput;
	StartGame();
}

// Start the game
// We first validate the data and then choose the appropriate state
void GameStarter::StartGame()
{
	cout << "Received input: " << input << endl;
	// Validate input and set appropriate state
	//ValidateInput validateInput = new ValidateInput();
	bool inputIsValid = true;
	if (inputIsValid)
		game.setState(GameLoop::getInstance());
	else
		game.setState(GameInvalid::getInstance());

	RunGameState();
}

// Start handling the game state
void GameStarter::RunGameState()
{
	//game.setState(GameOver::getInstance());
	game.run(input);
}
